src/plugins/navigation/public/top_nav_menu/top_nav_menu_data.tsx

```diff
@@ -50,6 +53,78 @@ export interface TopNavMenuData {
   type?: 'toggle' | 'button';
 }
 
-export interface RegisteredTopNavMenuData extends TopNavMenuData {
-  appName?: string;
+type RequireAtLeastOne<T, Keys extends keyof T = keyof T> = Pick<T, Exclude<keyof T, Keys>> &
+  {
+    [K in Keys]-?: Required<Pick<T, K>> & Partial<Pick<T, Exclude<Keys, K>>>;
+  }[Keys];
+
+interface TopNavMenuCommonData {
+  testId?: string;
+  className?: string;
+  disabled?: boolean | (() => boolean);
+  tooltip?: string | (() => string | undefined);
 }
+
+/* ToDo: Check with UX if this is needed
```

Comment:
anything tracking this?

Reply:
I will just ask UX.

Reply:
Spoke to UX; TopNavMenuLink is not something we like at the moment.